packages/core/src/interpreter.ts

```diff
@@ -369,6 +369,7 @@ export class Interpreter<
         listener(doneInvoke(this.id, doneData));
       }
       this._stop();
+      this._stopChildren();
```

Comment:
You might wonder why this can't be part of the `_stop` call as `_stop` gets called both:
- when reaching a final state
- when a machine gets stopped from the outside

The problem is that when `.stop()` (external call) gets called we first call `_stop` and then schedule an `xstate.stop` event. In this case, we should only stop children when we process that `xstate.stop` event and not immediately. This is mainly because exit actions should be able to send events to children. If we'd clear children in `_stop` then we would make this impossible.